plugins/orchestrator-backend/src/service/SonataFlowService.ts

```diff
@@ -180,12 +190,114 @@ export class SonataFlowService {
+      );
     } catch (error) {
       this.logger.error(`Error when fetching workflows: ${error}`);
     }
     return undefined;
   }
 
+  public async fetchWorkflowOverviews(): Promise<
+    WorkflowOverview[] | undefined
+  > {
+    try {
+      const response = await executeWithRetry(() =>
+        fetch(`${this.url}/management/processes`),
+      );
+
+      if (response.ok) {
+        const workflowIds = (await response.json()) as string[];
+        if (!workflowIds?.length) {
+          return [];
+        }
+        const items = await Promise.all(
+          workflowIds.map(async (workflowId: string) => {
+            const definition = await this.fetchWorkflowDefinition(workflowId);
+            if (!definition) {
+              return undefined;
+            }
+            let processInstances: ProcessInstance[] = [];
+            const limit = 10;
+            let offset: number = 0;
+
+            let lastTriggered: Date = new Date(0);
+            let lastRunStatus = '';
+            let counter = 0;
+            let totalDuration = 0;
+
+            do {
+              const graphQlQuery = `{ ProcessInstances(where: {processId: {equal: "${definition.id}" } }, pagination: {limit: ${limit}, offset: ${offset}}) { processName, state, start, lastUpdate, end } }`;
+
+              try {
+                const graphQlResponse = await executeWithRetry(() =>
+                  fetch(`${this.url}/graphql`, {
+                    method: 'POST',
+                    body: JSON.stringify({ query: graphQlQuery }),
+                    headers: { 'content-type': 'application/json' },
+                  }),
+                );
+
+                if (graphQlResponse.ok) {
+                  const json = await graphQlResponse.json();
+                  processInstances = json.data.ProcessInstances;
+                }
+              } catch (error) {
+                this.logger.error(`Error when fetching instances: ${error}`);
+              }
+
+              // eslint-disable-next-line no-loop-func
+              processInstances.forEach((pInstance: ProcessInstance) => {
+                if (new Date(pInstance.start) > lastTriggered) {
+                  lastTriggered = new Date(pInstance.start);
+                  lastRunStatus = pInstance.state;
+                }
+                if (pInstance.start && pInstance.end) {
+                  const start: Date = new Date(pInstance.start);
+                  const end: Date = new Date(pInstance.end);
+                  totalDuration += end.valueOf() - start.valueOf();
+                  counter++;
+                }
+              });
+              offset += limit;
+            } while (processInstances.length > 0);
+
+            const result: WorkflowOverview = {
+              workflowId: definition.id,
+              name: definition.name,
+              lastTriggered:
+                lastTriggered === new Date(0)
+                  ? ''
+                  : parseInt(
```

Comment:
for the sake of simplicity, I suggest to simply return to the UI lastTriggered.getTime() as a number
The UI will convert it to a pretty format

Reply:
+1

Reply:
eb735a3